modules/social_features/social_branding/src/Plugin/GraphQL/DataProducer/PlatformFeatureMachineName.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+namespace Drupal\social_branding\Plugin\GraphQL\DataProducer;
+
+use Drupal\social_graphql\Plugin\GraphQL\DataProducer\Entity\EntityDataProducerPluginBase;
+
+/**
+ * The machine name of the preferred feature.
+ *
+ * @DataProducer(
+ *   id = "platform_feature_machine_name",
+ *   name = @Translation("Platform Feature Machine Name"),
+ *   description = @Translation("The machine name of the preferred feature."),
+ *   produces = @ContextDefinition("string",
+ *     label = @Translation("Preferred Feature Machine Name")
+ *   ),
+ *   consumes = {
+ *     "preferredFeature" = @ContextDefinition("any",
+ *       label = @Translation("Preferred Feature"),
+ *       required = TRUE
+ *     )
+ *   }
+ * )
+ */
+class PlatformFeatureMachineName extends EntityDataProducerPluginBase {
+
+  /**
+   * Returns the machine name of the preferred feature.
+   *
+   * @param array $preferred_feature
+   *   The preferred feature.
+   *
+   * @return string
+   *   The machine name.
+   */
+  public function resolve(array $preferred_feature) : string {
+    return $preferred_feature['name'];
```

Comment:
Can we introduce a `PreferredPlatformFeature` class and interface here that contain `weight` and `name` as getters? That way we can use PHP's type-safety and editors can give hints about what data it contains (or should contain).

A good background read is here: https://steemit.com/php/@crell/php-never-type-hint-on-arrays

That way if we ever want to add a field we can see that we're probably going to break things (and provide a default value in the class).

Drupal uses a lot of structured arrays in its code but is trying to get away from that because it makes using PHPs type-system to avoid bugs nearly impossible (for example in [the overhaul of the render system](https://www.drupal.org/project/drupal/issues/1843798)).